### PR TITLE
Fix search bar props

### PR DESCRIPTION
**Pull request by bang88**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1859)

### CODE REVIEW

File: components/search-bar/index.native.tsx [hunk 1]
```diff
@@ -85,21 +85,22 @@ export default class SearchBar extends React.Component<ISearchBarNativeProps, Se
...
     const {
-      showCancelButton, cancelText, styles, value: propsValue,
+      showCancelButton, cancelText, styles = {}, value: propsValue,
       onChangeText, onChange, onSubmitEditing, disabled,
       ...restProps,
     } = this.props;
     const { style } = restProps;
     const { value, focus } = this.state;
     const _showCancelButton = showCancelButton || focus;
+    const _styles: ISearchBarStyle =  { ...SearchBarStyles, ...styles };
```
Comment (silentcloud) [inline components/search-bar/index.native.tsx]: 这个改动不符合 RN 自定义组件样式的规则了哈

https://github.com/ant-design/antd-mobile-samples/tree/master/rn-custom-ui

Comment (bang88) [reply components/search-bar/index.native.tsx]: 不太明白什么意思

Comment (silentcloud) [reply components/search-bar/index.native.tsx]: styles 在整个组件上下文是透传的，

Comment (bang88) [reply components/search-bar/index.native.tsx]: 明白了，那这里不需要改了。

**Pull request closed without merging**